src/vic3/data/alerts.rs

```diff
@@ -0,0 +1,125 @@
+use crate::block::Block;
+use crate::context::ScopeContext;
+use crate::db::{Db, DbKind};
+use crate::everything::Everything;
+use crate::game::GameFlags;
+use crate::item::{Item, ItemLoader};
+use crate::report::{err, ErrorKey};
+use crate::scopes::Scopes;
+use crate::token::Token;
+use crate::tooltipped::Tooltipped;
+use crate::trigger::validate_trigger;
+use crate::validate::validate_possibly_named_color;
+use crate::validator::Validator;
+
+#[derive(Clone, Debug)]
+pub struct Alert {}
+
+inventory::submit! {
+    ItemLoader::Normal(GameFlags::Vic3, Item::Alert, Alert::add)
+}
+
+impl Alert {
+    pub fn add(db: &mut Db, key: Token, block: Block) {
+        db.add(Item::Alert, key, block, Box::new(Self {}));
+    }
+}
+
+impl DbKind for Alert {
+    fn validate(&self, key: &Token, block: &Block, data: &Everything) {
+        let mut vd = Validator::new(block, data);
+
+        let loca = format!("alert_{key}_name");
+        data.verify_exists_implied(Item::Localization, &loca, key);
+        let loca = format!("alert_{key}_desc");
+        data.verify_exists_implied(Item::Localization, &loca, key);
+        let loca = format!("alert_{key}_hint");
+        data.verify_exists_implied(Item::Localization, &loca, key);
+        let loca = format!("alert_{key}_action");
+        data.verify_exists_implied(Item::Localization, &loca, key);
+        let loca = format!("{key}_setting_name");
+        data.verify_exists_implied(Item::Localization, &loca, key);
+
+        vd.req_field("script_context");
+        let mut sc = ScopeContext::new(Scopes::None, key);
+        if let Some(script_context) = vd.field_value("script_context") {
+            let mut found = false;
+            for &(context, s) in SCRIPT_CONTEXTS {
```

Comment:
Might be better as a FnvHashMap for uniformity but not necessary

Reply:
I considered it but this array is used so rarely, comparatively, that it didn't seem worth the extra code